### DISCUSSION TITLE
How to build glad2?

### DISCUSSION THREAD

**Discussion by apivovarov** [Dav1dde/glad#432] [Q&A] [answered]
I tried the following
```
cmake -B build -S cmake

cmake --build build
```
No errors, but the build process did nothing

`make install` does not work too.

conan2 and vcpkg have 0.1.36 only.

How to build and install glad2 to linux/mac?

**Comment by apivovarov** [comment, marked as answer]
ok, looks like I need to use [webservice for glad2](https://glad.sh/) to generate `glad.zip` and use `include/glad/gl.h` and `src/gl.c` from the zip.

**Reply by Dav1dde** [reply]
The easiest way is to just use the web generator or run glad locally when you need to and copy the artifacts to your project (like you did).

Alternatively you can use the CMake files in the git repo

**Reply by apivovarov** [reply]
Would be nice to have an example on how exactly we can use cmake in glad2.

**Reply by Dav1dde** [reply]
There are examples in the repo, but I don't recommend you to use cmake, just generate once and add the files to your repository. It gives you nice deltas if you choose to re-generate, faster builds and less dependencies.

**Reply by apivovarov** [reply]
ok, I see [c++/multiwin_mx/CMakeLists.txt](https://github.com/Dav1dde/glad/blob/glad2/example/c%2B%2B/multiwin_mx/CMakeLists.txt) in the examples. 
Is it possible to generate `gl.h` and `gl.c` locally using CMake or some script instead of using the website?

**Reply by Dav1dde** [reply]
Yes, you can just clone the repository and run glad via `python -m glad` or install it through [pip](https://pypi.org/project/glad2/) `pip install --user glad2`.

If you check the header of your generated files (through the website) the first comment will actually include the entire commandline which you can use the generate the same files:

```
/**
 * Loader generated by glad 2.0.4 on Fri Jun 30 17:56:55 2023
 *
 * [snip...]
 * 
 * Commandline:
 *    --api='gl:compatibility=4.5' --extensions='GL_AMD_conservative_depth,GL_AMD_debug_output,GL_AMD_depth_clamp_separate' c
 */
```

**Reply by apivovarov** [reply]
my `gl.h` shows 
```
* Commandline:
*    --api='gl:core=3.3' --extensions='' c
 ```

I guess the command should be
```
python3 -m glad --api='gl:core=3.3' --extensions='' --out-path . c
```
